bezier/src/cubic_bezier.rs

```diff
@@ -183,6 +183,140 @@ impl CubicBezierSegment {
         find_cubic_bezier_inflection_points(self)
     }
 
+    /// Return local x extremum points or None if this curve is monotone.
+///
+/// This returns the advancements along the curve, not the actual x position.
+    pub fn find_x_extremums(&self) -> UpToTwo<f32> {
+        let mut ret = UpToTwo::new();
+        // See www.faculty.idc.ac.il/arik/quality/appendixa.html for an explanation
+        // The derivative of a cubic bezier curve is a curve representing a second degree polynomial function
+        // f(x) = a * x² + b * x + c such as :
+        let a = 3.0 * (self.to.x - 3.0 * self.ctrl2.x + 3.0 * self.ctrl1.x - self.from.x);
+        let b = 6.0 * (self.ctrl2.x - 2.0 * self.ctrl1.x + self.from.x);
+        let c = 3.0 * (self.ctrl1.x - self.from.x);
+
+        println!("a : {} | b : {} | c : {}", a, b, c);
+
+        // If the derivative is a linear function
+        if a == 0.0 {
+            if b == 0.0 {
+                // If the derivative is a constant function
+                if c == 0.0 {
+                    ret.push(0.0);
+                }
+            } else {
+                ret.push(-c / b);
+            }
+            return ret;
+        }
+
+        fn in_range(t: f32) -> bool { t >= 0.0 && t < 1.0 }
```

Comment:
Let's use t strictly superior to zero here, since this is primarily useful to split the curve into monotone segments and computing the bounding rectangle of the curve, and in both situations there is no need to special-case of split the curve at t = 0.0.